client/my-sites/email/email-management/home/email-no-domain.tsx

```diff
@@ -1,25 +1,50 @@
 import { isFreePlan } from '@automattic/calypso-products';
 import { localize } from 'i18n-calypso';
-import PropTypes from 'prop-types';
 import { useSelector } from 'react-redux';
 import Illustration from 'calypso/assets/images/domains/domain.svg';
 import EmptyContent from 'calypso/components/empty-content';
-import { recordInboxUpsellTracksEvent } from 'calypso/my-sites/email/email-management/home/utils';
+import TrackComponentView from 'calypso/lib/analytics/track-component-view';
+import { recordEmailUpsellTracksEvent } from 'calypso/my-sites/email/email-management/home/utils';
 import { hasDomainCredit } from 'calypso/state/sites/plans/selectors';
+import { SiteData } from 'calypso/state/ui/selectors/site-data';
+import type { AppState } from 'calypso/types';
 
-const noop = () => {};
-
-const EmailNoDomain = ( { selectedSite, translate, source } ) => {
-	const hasAvailableDomainCredit = useSelector( ( state ) =>
+const EmailNoDomain = ( {
+	selectedSite,
+	translate,
```

Comment:
I came to think of one more thing here! 💡 

Given that this is a functional component, we could use the `useTranslate` hook rather than the `localize` higher order component (which is where the `translate` prop is injected right now). This way, we wouldn't have to manually define a type for the `translate` prop. That could be nice, because in reality, `translate` can return other types than just `string`.

Reply:
@fredrikekelund I have made the update! Mind giving one last look before I deploy?